pkg/cloudprovider/cloudprovider.go

```diff
@@ -100,6 +100,32 @@ func New(ctx awscontext.Context) *CloudProvider {
 	}
 }
 
+func (c *CloudProvider) GetDriftedNodes(ctx context.Context, provisioner *v1alpha5.Provisioner, nodes []v1.Node) []v1.Node {
+	driftedNodesMap := map[string]bool{}
+	var driftedNodes []v1.Node
+	//Consider if provisioner should be a part of this interface
```

Comment:
I think this is a good direction to be thinking about, but an in-line interface here might be setting up too much structure to the code that we may not need in the future.

Is it possible to restructure this code as a `IsNodeDrifted()` so that we just simply return a boolean? This way we can leave getting any node objects to the controller that wants to call this function, and this helper function can be a way to get the information.